.github/workflows/CI-distro.yml

```
@@ -0,0 +1,108 @@
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  generate-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      gap-packages: ${{ steps.set-matrix.outputs.gap-packages }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: "Set up Julia"
+        uses: julia-actions/setup-julia@v2
+        with:
+          version: '1'
+      - name: "Cache artifacts"
+        uses: julia-actions/cache@v2
+      - name: "Build package"
+        uses: julia-actions/julia-buildpkg@v1
+      - name: Get list of GAP packages
+        id: set-matrix
+        run: julia --project=. -e '
+                  using Artifacts, TOML;
+                  output = sprint(print, "gap-packages=", map(name -> name[9:end], sort!(collect(filter(startswith("GAP_pkg_"), keys(TOML.parsefile(find_artifacts_toml(Base.active_project()))))))));
+                  println(output);
+                  open(ENV["GITHUB_OUTPUT"], "a") do io;
+                    println(io, output);
+                  end;'
+
+  test:
+    name: ${{ matrix.gap-package }}
+    needs: generate-matrix
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 20
+    continue-on-error: ${{ matrix.julia-version == 'nightly' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        julia-version:
+          - '1'
+        julia-arch:
+          - x64
+        os:
+          - ubuntu-latest
+        gap-package: ${{fromJSON(needs.generate-matrix.outputs.gap-packages)}}
+        exclude:
+          - gap-package: 'alnuth'                     # `AL_EXECUTABLE, the executable for PARI/GP, has to be set`
+          #- gap-package: 'autodoc'                    # test failure
+          #- gap-package: 'caratinterface'             # `Carat program Bravais_inclusions failed with error code 1`
+          - gap-package: 'curlinterface'              # no jll
+          - gap-package: 'example'                    # no jll
+          #- gap-package: 'examplesforhomalg'          # `Error, found no GAP executable in PATH`
+          #- gap-package: 'gapdoc'                     # test failure
+          - gap-package: 'guarana'                    # `AL_EXECUTABLE, the executable for PARI/GP, has to be set`
+          - gap-package: 'hap'                        # `polymake command not found. Please set POLYMAKE_COMMAND by hand`
+          - gap-package: 'hapcryst'                   # `polymake command not found. Please set POLYMAKE_COMMAND by hand`
+          #- gap-package: 'io'                         # test failure
+          - gap-package: 'itc'                        # dependency `xgap` has no jll
+          #- gap-package: 'localizeringforhomalg'      # `Error, found no GAP executable in PATH`
+          #- gap-package: 'normalizinterface'          # `Error, Some error in the normaliz input data detected: Cone Property in last line not allowed for lattice ideals`
+          #- gap-package: 'packagemanager'             # test failure
```

this is weird. looking at why PackageManager tests fail, first is this:
```
LoadPackage("curlInterface", false);
# Expected output:
true
# But found:
fail
```
That's clear enough and is waiting for https://github.com/JuliaPackaging/Yggdrasil/pull/9999

But then:
```
# Input is:
InstallPackage("https://github.com/gap-packages/io.git");
# Expected output:
true
# But found:
#I  PackageInfo.g validation failed
#I  There may be problems with the package
true
########
```
I suspect what happens is that it fails to build the manual, or at least the PDF part of the manual, due to missing latex. Then you'd get something like this:
```
gap> ValidatePackageInfo("PackageInfo.g");
#E  component `PDFFile' must be bound to a string denoting a relative path to a readable file
false
gap> ValidatePackageInfo("PackageInfo.g" : quiet);
false
```

So to fix this, we'd have to install `latex` on the runner. That would probably also help GAPDoc and perhaps autodoc.

---

I think now only the missing `curlInterface` is an issue
```suggestion
          - gap-package: 'packagemanager'             # tests need curlInterface to passe, reactivate once we have a working JLL for that
```